internal/resolver/resolver.go

```diff
@@ -71,11 +72,22 @@ func (r *Resolver) Resolve(reqs []*chart.Dependency, repoNames map[string]string
 			return nil, errors.Wrapf(err, "dependency %q has an invalid version/constraint format", d.Name)
 		}
 
-		idx := filepath.Join(r.cachepath, helmpath.CacheIndexFile(repoNames[d.Name]))
+		repoName := repoNames[d.Name]
+		// if the repository was not defined, but the dependency defines a repository url, bypass the cache
+		if repoName == "" && d.Repository != "" {
+			fmt.Println(d.Name)
+			fmt.Println(d.Repository)
```

Comment:
Were these print statements debugging that was accidentally left in?

Reply:
Oops, yes they were. Good catch.